File: src/ObjectManager/Object.Tes/FilePacks/Records/005-ARMA.Armature (Model).cs
﻿using OA.Core;

namespace OA.Tes.FilePacks.Records
{
    public class ARMARecord : Record
    {
        public override string ToString() => $"ARMA: {EDID.Value}";
        public STRVField EDID { get; set; } // Editor ID

        public override bool CreateField(UnityBinaryReader r, GameFormatId format, string type, int dataSize)
        {
            switch (type)
            {
                case "EDID": EDID = r.ReadSTRV(dataSize); return true;
                default: return false;
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/005-ARMA.Armature (Model).swift
//
//  ARMARecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class ARMARecord: Record {
    public override var description: String { return "ARMA: \(EDID)" }
    public var EDID: STRVField = STRVField_empty // Editor ID
    
    override func createField(_ r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        switch type {
        case "EDID": EDID = r.readSTRV(dataSize)
        default: return false
        }
        return true
    }
}
